### PR TITLE
CreateMockAPI

### PR DESCRIPTION
**Pull request by tumbledwyer**
This PR enables the use of mock data so that the backend does not have to be started for dev to continue on the frontend.

Axios is mocked out so that all fetching from components works in the same way it would in production. 

This can probably be removed in future once backend development has caught up to the frontend.

Set `REACT_APP_MOCK_BACKEND=true` in `.env.local` to mock the backend

### CODE REVIEW

**Review by marrouchi — APPROVED** [review]

File: src/services/mockData.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  gender: 'Male',
+  dob: new Date(1605522215666),
+  phoneNumber: '085145855',
+  city: 'Welkom',
+  updatedBy: undefined
+}
+
+const goldenRecord: PatientRecord = {
+  auxId: '0x77',
+  type: 'Current',
```
Comment (marrouchi) [inline src/services/mockData.ts:34]:
```suggestion
  type: 'Golden',
```
right ?

File: src/services/mockData.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    patient_id: '0x5a',
+    state: NotificationState.New,
+    golden_id: '0x45',
+    score: 0.5
+  }
+]
+
+const patientRecord: PatientRecord = {
+  auxId: '0x66',
+  type: 'Current',
```
Comment (marrouchi) [inline src/services/mockData.ts:19]:
```suggestion
  type: 'Candidate',
```
Should this be candidate?

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: src/services/mockBackend.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+const moxios = axios.create()
+
+const axiosMockAdapterInstance = new AxiosMockAdapter(moxios, {
+  delayResponse: 0
+})
+
+const { notifications, patientRecord, goldenRecords } = mockData
+
+axiosMockAdapterInstance
+  .onGet('/MatchesForReview')
```
Comment (marrouchi) [inline src/services/mockBackend.ts:15]:
```suggestion
  .onGet(ROUTES.GET_NOTIFICATIONS)
```
How about we re-use ROUTES

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: src/services/mockData.ts (hunk 2 above)
Comment (tumbledwyer) [reply src/services/mockData.ts:19]: This one is definitely current as it is the current patient. Candidates are potential golden records to link to

File: src/services/mockData.ts (hunk 1 above)
Comment (tumbledwyer) [reply src/services/mockData.ts:34]: This one should actually be empty as it doesn't come from the backend, it's calculated on the front end. All that logic needs to move to the notification endpoint though.

File: src/services/mockBackend.ts (hunk 3 above)
Comment (tumbledwyer) [reply src/services/mockBackend.ts:15]: Yeah, I was thinking about that actually. I'll put out a const class

**Review by marrouchi — APPROVED** [review]
Looks good !

**Pull request merged**